x/recovery/keeper/ibc_callbacks.go

```diff
@@ -64,49 +65,44 @@ func (k Keeper) OnRecvPacket(
 		)
 	}
 
-	// case 1: sender ≠ recipient.
-	// Recovery is only possible for addresses in which the sender = recipient
-	// (i.e transferring to your own account in Evmos).
+	// Check if sender != recipient, as recovery is only possible for transfers to
+	// a sender's own account on Evmos (sender == recipient)
 	if !sender.Equals(recipient) {
 		// Continue to the next IBC middleware by returning the original ACK.
 		return ack
 	}
 
-	// get the sender account
-	account := k.accountKeeper.GetAccount(ctx, sender)
+	// get the recipient/sender account
+	account := k.accountKeeper.GetAccount(ctx, recipient)
 
 	// recovery is not supported for vesting or module accounts
-	_, isVestingAcc := account.(vestexported.VestingAccount)
-	if isVestingAcc {
+	if _, isVestingAcc := account.(vestexported.VestingAccount); isVestingAcc {
 		return ack
 	}
 
-	_, isModuleAccount := account.(authtypes.ModuleAccountI)
-	if isModuleAccount {
+	if _, isModuleAccount := account.(authtypes.ModuleAccountI); isModuleAccount {
 		return ack
 	}
 
-	// Case 2. sender pubkey is a supported key (eth_secp256k1, amino multisig, ed25519)
-	// ==> Continue and return success ACK as the funds are not stuck on chain
-	if account != nil &&
-		account.GetPubKey() != nil &&
+	// Check if recipient pubkey is a supported key (eth_secp256k1, amino multisig,
+	// ed25519). Continue and return success ACK as the funds are not stuck on
+	// chain for supported keys
+	if account != nil && account.GetPubKey() != nil &&
 		evmos.IsSupportedKey(account.GetPubKey()) {
```

Comment:
let's remove the `account.GetPubKey() != nil` by returning false in `IsSupportedKey` when the pubkey is `nil`

Reply:
```suggestion
	if account != nil &&
		evmos.IsSupportedKey(account.GetPubKey()) {
```